### PR TITLE
Make NML import more performant and resilient

### LINKED ISSUE #4374: Make import of NMLs more resilient
To make it easier to switch from Knossos to WK, the import of NMLs should support:

-  unconnected components (the unconnected components should be split into different trees (which then are grouped together using our group feature). See this script: https://github.com/scalableminds/wknml/blob/master/examples/fix_unlinked_nml.py)
- `<things>` without names (can default to "Unnamed tree 1" or similar) (already supported by the backend)

Might need to happen in front-end in back-end.

### LINKED ISSUE #4548: Frontend NML parser: show error details in toast
In the toast I only get "x.nml could not be parsed as either NML or protobuf."
It would be nice if error summaries from the different parsing attempts, as are already present in the console, could also be included in the toast (or at least “press f12 for error details in the browser console” or something like that)

### PR DESCRIPTION
**Pull request by daniel-wer**
- Better performance, by keeping trees and groups mutable until put into the store (From ~5s to ~1.5s for this part for a 200,000 node tracing)
- Allow missing tree names and assign a new tree name for missing and empty tree names
- Align tree timestamp assignment to backend NML parser. Replace invalid timestamps with a default timestamp of 0.
- Fixed that validation errors are shown for NML-like files
- Split unconnected trees during NML import and wrap them in a group.

Ideally, I'd like to avoid the redefinition of the `MutableTree` type and, therefore, experimented with recursive read-only instructions for flow, but I couldn't get those to work for the diffable maps :/
(for reference, the most success I had was with the solution outlined in https://github.com/facebook/flow/issues/5844#issuecomment-407169272)

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- [knossos_nmls.zip](https://github.com/scalableminds/webknossos/files/4504148/knossos_nmls.zip)
- Import these knossos NMLs in the tracing view.
  - Unconnected trees should be split
  - There should be no errors during saving
- Try to import an NML where the trees have no name or the nodes have no attributes (except for position and id)
- Check that validation errors are shown for NML-like files, e.g. by removing the id attribute of one of the nodes before upload.

### Issues:
- contributes to #4374 (needs to be done in backend as well)
- fixes #4548 

------
- [x] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)
- [ ] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable
- [ ] Updated [documentation](../blob/master/docs) if applicable
- [ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes
- [ ] Needs datastore update after deployment
- [x] Ready for review

### CODE REVIEW

File: frontend/javascripts/oxalis/model/helpers/nml_helpers.js [hunk 1]
```diff
@@ -356,52 +371,121 @@ function _parseBool(obj: Object, key: string, defaultValue?: boolean): boolean {
...
+    children: [],
+  };
+
+  const newTrees = [];
+  for (let i = 0; i < components.length; i++) {
+    const nodeIds = components[i];
+
+    const edges = nodeIds.reduce(
+      // Only consider outgoing edges as otherwise each edge would be collected twice
+      (agg, nodeId) => agg.concat(tree.edges.getOutgoingEdgesForNode(nodeId)),
```
[the remark this answers is not in the document]
Comment (daniel-wer) [reply frontend/javascripts/oxalis/model/helpers/nml_helpers.js:446]: Good point, you would have to return agg explicitly then, though, as push doesn't return the array but the inserted value. Anyway, I found out that `<array>.flatMap(...)` is now widely supported (https://caniuse.com/#feat=array-flat), so I'll use that :)

[the remark this answers is not in the document]
Comment (philippotto) [reply frontend/javascripts/oxalis/model/helpers/nml_helpers.js:446]: Ah, even better :)

**Comment by daniel-wer** [comment]
Thanks for the review, I updated flow to a somewhat newer version, because the `flatMap` type definition was missing - fixed the new errors as good as possible. The newest flow version adds lots of new errors, so that'll be a project for another day ^^

**Comment by daniel-wer** [comment]
@philippotto Could you have a look at my new changes and maybe test whether `yarn flow test` works for you locally with this version? :)

**Comment by philippotto** [comment]
> @philippotto Could you have a look at my new changes and maybe test whether `yarn flow test` works for you locally with this version? :)

After I upgraded node to v12, everything worked fine :) Maybe post a PSA in slack after this is merged?

**Comment by daniel-wer** [comment]
Absolutely, will do, thanks for testing :)

**Pull request merged**